doc/source/train/examples/pytorch/distributing-pytorch/README.ipynb

```diff
@@ -0,0 +1,599 @@
+   "metadata": {},
+   "source": [
+    "## Step 2: Distribute training to multiple machines with Ray Train\n",
+    "\n",
+    "Let’s modify this example to run with Ray Train on multiple machines. Essentially, Ray Train allows you to wrap your PyTorch training code in a function and run that function on each of the workers in your Ray Cluster. With just a few modifications, you get all the fault tolerance and auto-scaling of a [Ray Cluster](https://docs.ray.io/en/latest/cluster/getting-started.html), as well as the observability and ease-of-use of [Ray Train](https://docs.ray.io/en/latest/train/train.html).\n",
+    "\n",
+    "First, import a few Ray Train modules and enable [Ray Train v2](https://github.com/ray-project/enhancements/blob/main/reps/2024-10-18-train-tune-api-revamp/2024-10-18-train-tune-api-revamp.md)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "# This will be removed once Ray Train v2 is part of latest Ray version\n",
+    "echo \"RAY_TRAIN_V2_ENABLED=1\" > /home/ray/default/.env"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Load env vars in notebooks\n",
+    "from dotenv import load_dotenv\n",
+    "load_dotenv()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ray.train\n",
+    "from ray.train import RunConfig, ScalingConfig\n",
+    "from ray.train.torch import TorchTrainer\n",
+    "\n",
+    "import tempfile"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, modify the training function you wrote earlier. Every difference from the previous script is highlighted and explained with a numbered comment; for example, “[1].”"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def train_func_per_worker(config: Dict):\n",
+    "    lr = config[\"lr\"]\n",
+    "    epochs = config[\"epochs\"]\n",
+    "    batch_size = config[\"batch_size_per_worker\"]\n",
+    "\n",
+    "    # Get dataloaders inside the worker training function\n",
+    "    train_dataloader, test_dataloader = get_dataloaders(batch_size=batch_size)\n",
+    "\n",
+    "    # [1] Prepare Dataloader for distributed training\n",
+    "    # The prepare_data_loader method shards the dataset among workers and\n",
+    "    # moves batches to the correct device\n",
+    "    # ===================================================================\n",
+    "    train_dataloader = ray.train.torch.prepare_data_loader(train_dataloader)\n",
+    "    test_dataloader = ray.train.torch.prepare_data_loader(test_dataloader)\n",
```

Comment:
* `"Shard the datasets among workers and move batches to the correct device" -> "Assign unique rows of data to each worker so that the model sees each row once"`
    * And "NOTE: This only works for map-style datasets. For a general distributed preprocessing and sharding solution, see the next part using Ray Data for data ingestion."